packages/core/admin/admin/src/components/FormInputs/Boolean.tsx

```diff
@@ -8,53 +8,38 @@ import { useField } from '../Form';
 
 import { InputProps } from './types';
 
-const BooleanInput = forwardRef<HTMLInputElement, InputProps>(
-  ({ disabled, label, hint, name, required }, ref) => {
-    const { formatMessage } = useIntl();
-    const field = useField<boolean | null>(name);
-    const fieldRef = useFocusInputField(name);
+const BooleanInput = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
+  const { formatMessage } = useIntl();
+  const field = useField<boolean | null>(props.name);
+  const fieldRef = useFocusInputField(props.name);
 
-    const composedRefs = useComposedRefs<HTMLInputElement | null>(ref, fieldRef);
+  const composedRefs = useComposedRefs<HTMLInputElement | null>(ref, fieldRef);
 
-    return (
-      <ToggleInput
-        ref={composedRefs}
-        checked={field.value === null ? null : field.value || false}
-        disabled={disabled}
-        hint={hint}
-        // @ts-expect-error – label _could_ be a ReactNode since it's a child, this should be fixed in the DS.
-        label={label}
-        error={field.error}
-        /**
-         * TODO: reintroduce labelActions
-         */
-        // labelAction={labelAction}
-        name={name}
-        offLabel={formatMessage({
-          id: 'app.components.ToggleCheckbox.off-label',
-          defaultMessage: 'False',
-        })}
-        onLabel={formatMessage({
-          id: 'app.components.ToggleCheckbox.on-label',
-          defaultMessage: 'True',
-        })}
-        onChange={field.onChange}
-        required={required}
-        onClear={() => {
-          field.onChange(name, null);
-        }}
-        // TODO: re-introduce clear label
-        // clearLabel={
-        //   isNullable
-        //     ? formatMessage({
-        //         id: 'app.components.ToggleCheckbox.clear-label',
-        //         defaultMessage: 'Clear',
-        //       })
-        //     : undefined
-        // }
-      />
-    );
-  }
-);
+  return (
+    // @ts-expect-error – label _could_ be a ReactNode since it's a child, this should be fixed in the DS.
+    <ToggleInput
+      ref={composedRefs}
+      checked={field.value === null ? null : field.value || false}
+      error={field.error}
+      /**
+       * TODO: reintroduce labelActions
+       */
+      // labelAction={labelAction}
+      offLabel={formatMessage({
+        id: 'app.components.ToggleCheckbox.off-label',
+        defaultMessage: 'False',
+      })}
+      onLabel={formatMessage({
+        id: 'app.components.ToggleCheckbox.on-label',
+        defaultMessage: 'True',
+      })}
+      onChange={field.onChange}
+      onClear={() => {
+        field.onChange(props.name, null);
+      }}
+      {...props}
```

Comment:
I really like the idea of passing the props with the spread operator and leaving it as the final prop in the Component, I am wondering if we can use the same order in the other components like Email and Password

Reply:
we can't because the type will be overwritten breaking the inputs :)